src/main/java/software/amazon/awssdk/crt/http/Http2ConnectionSettings.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.awssdk.crt.http;
+
+import java.util.List;
+import java.util.ArrayList;
+
+public class Http2ConnectionSettings {
+
+    private List<Http2ConnectionSetting> settings;
```

Comment:
I don't understand why we have this class?
it's just a list under the hood, so why not just have the list
with the wrapper class now it's weird like "where are the getters" and "calling the same function twice adds to values I didn't expect that"

maybe make this a helper, like:
```
connection.changeSettings(new Http2ConnectionSetting.Builder()
                  .enablePush(false)
                  .build()
```